src/fides/api/util/consent_util.py

```diff
@@ -228,3 +247,281 @@ def get_or_create_fides_user_device_id_provided_identity(
+
+
+def validate_privacy_notice_dry_update(dry_update: PrivacyNotice) -> None:
+    """
+    Verify that a dry update of a PrivacyNotice satisfies the constraints
+    for creating a privacy notice.
+
+    This is done here instead of upfront because we need access to the current values
+    of the privacy notice in the database combined with the patch updates from the request
+    """
+    try:
+        PrivacyNoticeCreation.from_orm(dry_update)
+    except ValueError as exc:
+        raise HTTPException(
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            # pylint: disable=no-member
+            detail=exc.errors(),  # type: ignore
+        )
+
+
+def prepare_privacy_notice_patches(
+    privacy_notice_updates: List[PrivacyNoticeWithId],
+    db: Session,
+    model: Union[Type[PrivacyNotice], Type[PrivacyNoticeTemplate]],
+) -> List[
+    Tuple[PrivacyNoticeWithId, Optional[Union[PrivacyNotice, PrivacyNoticeTemplate]]]
+]:
+    """
+    Prepares PrivacyNotice or Template upserts including performing data use
+    conflict validation on proposed changes.
+
+    For Privacy Notices, returns a list of tuples with the update data alongside the existing Privacy
+    db record that will be updated.
+
+    For PrivacyNoticeTemplates, returns a list of tuples with the data to be upserted, alongside
+    the existing PrivacyNoticeTemplate if it exists or None otherwise.
+
+    :param privacy_notice_updates: List of privacy notice schemas with ids: appropriate
+    for editing PrivacyNotices or upserting PrivacyNoticeTemplates
+    :param db: Session
+    :param model: one of PrivacyNotice or PrivacyNoticeTemplate
+    :return:
+    """
+    allow_create: bool = model == PrivacyNoticeTemplate
+    ignore_disabled: bool = model != PrivacyNoticeTemplate
+
+    # first we populate a map of privacy notices or templates in the db, indexed by ID
+    existing_notices: Dict[str, Union[PrivacyNotice, PrivacyNoticeTemplate]] = {}
+    for existing_notice in model.query(db).all():
+        existing_notices[existing_notice.id] = existing_notice
+
+    # then associate existing notices/templates with their creates/updates
+    # we'll return this set of data to actually process updates
+    updates_and_existing: List[
+        Tuple[
+            PrivacyNoticeWithId, Optional[Union[PrivacyNotice, PrivacyNoticeTemplate]]
+        ]
+    ] = []
+    for update_data in privacy_notice_updates:
+        update_data = transform_fields(
+            transformation=escape,
+            model=update_data,
+            fields=PRIVACY_NOTICE_ESCAPE_FIELDS,
+        )  # type: ignore
+        if update_data.id not in existing_notices:
+            if not allow_create:
+                raise HTTPException(
+                    status_code=HTTP_404_NOT_FOUND,
+                    detail=f"No {model.__class__.__name__} found for id {update_data.id}.",
+                )
+            updates_and_existing.append((update_data, None))
```

Comment:
New capability that is for templates only - we can stage a template for creation here.